### PR TITLE
treat buffer respecting size (1)

### PR DESCRIPTION
**Pull request by lowgear**
Buffers of machine_msg_t were treated as c-string in multiple places.

### CODE REVIEW

**Comment by lowgear** [comment]
These are not all the changes required, but further changes will be introduced in next PR.

File: test/odyssey/test_scram_frontend.sh [hunk 1]
```diff
@@ -5,22 +5,22 @@ if test $PGVERSION -lt 10; then
     exit 0
 fi
 
-PGPASSWORD=incorrect_password psql -h $ODYSSEY_HOST -p $ODYSSEY_PORT -U frontend_auth_plain -c "SELECT 1" scram_db > /dev/null 2>&1 && {
-    echo "ERROR: successfully auth with incorrect password and plain password in config"
-    exit 1
-}
+#PGPASSWORD=incorrect_password psql -h $ODYSSEY_HOST -p $ODYSSEY_PORT -U frontend_auth_plain -c "SELECT 1" scram_db > /dev/null 2>&1 && {
```
Comment (x4m) [inline test/odyssey/test_scram_frontend.sh:8]: Don't we need these tests?

Comment (lowgear) [reply test/odyssey/test_scram_frontend.sh:8]: Added this by mistake. I commented this for debug.

**Comment by x4m** [comment]
Thanks, pushed!

**Pull request merged**